### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Iterate Innovate Run, Performance Dashboard

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#82596]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Sorry for the slow feedback on this! There was a lot I wanted to write up!
- Just to reiterate something I said in the meeting: making accessible data visualizations is *hard*. The biggest reason is that a data visualization can serve multiple purposes simultaneously --- some users will look at a chart for exact or approximate numerical values, while others will look for "at a glance" trends. Capturing both of these interactions in an intelligible way without overwhelming the user, and doing it in any sort of automated way, is really hard!

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [ ] **Engineering note:** Follow the [accessibility guidance provided by Chart.js](https://www.chartjs.org/docs/latest/general/accessibility.html), namely that each chart should have an accessible name set by either:
   - Giving the `canvas` a `role="img"` and an `aria-label`; or,
   - Placing accessible fallback content inside the canvas (eg. `<canvas ... ><p>fallback</p></canvas>`).
   
   Between these two options, I have a slight preference for `role="img"` plus `aria-label`, because I'm a little more confident in how screen readers will announce these. However, if you end up making any of the parts of the `canvas` interactive (eg. tooltips displayed on hover) then I'm not sure that `role="img"` is a good idea. If you go down that road, try the fallback content approach and then please ping me in Slack so I can do some tests and give you some interactive chart accessibility guidance.
- [ ] **Must:** Whenever possible, the accessible name of the `canvas` should describe the general trend of the data displayed in the charts you've designed. An example might be "Submissions increasing from 163,000 to 185,000 with month-to-month variation." If that kind of description isn't possible --- either because it's not programmatically determinable, or because the chart is too complex to describe concisely --- then the accessible name should describe what kind of information is in the chart, eg. "Bar chart showing logged in verified users by month, broken down into new and existing sessions" or something similar.
- [ ] **Must:** Whenever possible, provide a link immediately after the chart to the raw data to allow users to explore that data to better understand the information. Linking to a CSV file is acceptable here. Note that the link should be descriptive, eg. "Data set for signed-in verified sessions (CSV)" or similar. Definitely include the file type in the link text.
- [ ] **Must:** Don't use color as the only way of differentiating information. (Technically, exceptions exist for sufficiently different luminosities, but I think you'll have trouble achieving that with the number of colors you've got in some of these charts.) Two useful techniques for distinguishing between elements in the charts:
   - Use patterns or textures, eg. lines, dots, etc. This should be sufficient for the bar charts in the prototype. Make sure that the chart's key includes that pattern in the color sample.
   - Use shapes, eg. square, circle, triangle, etc. This should work for the satisfaction rating line chart. Likewise, make sure the key includes the shape.
- [ ] **Must:** Whenever you have arrow icons indicating up or down trends, those icons should have `alt` text (if coded as an `img`) or an `aria-label` (if coded as a `span` or `i` etc) --- or, instead, those icons should be hidden from screen readers entirely using `aria-hidden="true"` and you should include screen reader only text immediately after the icon (eg. `<i ... aria-hidden="true" /><span class="sr-only">Increase of</span> 6.3%`). All of these are basically equivalent, no strong preference from me on how you choose to do this.
- [ ] **Should:** I think your color choices for red and green text look okay, but double check them using the [WebAIM contrast checker](https://webaim.org/resources/contrastchecker/). Would be great to pass WCAG AAA, but must absolutely pass WCAG AA.
- [ ] **Should:** I think some additional context is needed for the Download PDF and Share Link buttons. As presented here, it's not clear what actually happens when you click Share Link, and it's not clear what the scope of Download PDF is (all data vs. data currently on screen). A little text explaining these buttons will go a long way.
- [ ] **Must:** When the Share Link button copies the link URL to the clipboard, some kind of feedback needs to be provided to indicate that the action has occurred. My recommendation would be a success alert (maybe a slim alert, though I'd check with @allison0034 on that usage), with the alert receiving keyboard focus so that it's announced immediately by screen readers. There are other options that could work here too --- happy to talk through any ideas you have.
- [ ] **Should:** I would highly recommend moving the Share Link and Download PDF buttons into the main column of content rather than where they are now, floating off to the right. Users with screen magnifiers (either software or physical devices on their monitors) typically focus on the main column of content and may not notice anything over to the right if there's not a clear visual indicator to draw their attention that direction.
- [ ] **Must:** I'll echo @erinrwhite's feedback --- Some additional headings could provide better structure to both the highlights and details tabs. Likewise, those headings should also describe the timeframe currently displayed.
- [ ] **Must:** Echoing @erinrwhite again --- I think a filter approach for timeframe would work better here than just toggling with radio buttons. Ideally, I'd like to see the "Show all benefit types" dropdown on the details tab to be grouped in with the timeframe filter. The actual filter action should be initiated by a button separate from the options, eg. `<button>Filter</button>` or similar.
- [ ] **Must:** Sortable tables are tricky! My go-to resource here is from Adrian Roselli, [Sortable Table Columns](https://adrianroselli.com/2021/04/sortable-table-columns.html).
- [ ] **Should:** In the Benefit Type column of the details table, try to avoid truncating text as much as possible.
- [ ] **Must:** In the Sat Score column of the details table, each "Read reviews" link will need to have a unique accessible name so that they can be easily differentiated for screen reader users. Using an `aria-label` or `sr-only` text is fine here, eg. `<a href="...">Read reviews<span class="sr-only"> for Disability - Check VA claim or appeal status</span></a>`.
- [ ] **Must:** The details table should have a `caption` element, which functions as the title of the table.
- [ ] **Consider:** Shane suggested zebra striping the table rows, and I think that's a great idea. If not striped, consider adding a bottom border to each row.

Whew! That's a lot! Please don't hesitate to ping me in Slack if you have anything you want to talk through (synchronously or async).

## Governance team actions
- [ ] Format feedback as individual tasks (check boxes)
- [ ] Assign this ticket to the VFS team member that opened the Slack request
- [ ] Add the VFS team product label
- [ ] Add the VFS team the feature label (if applicable)
- [ ] Add the touchpoint labels
- [ ] Add the practice area labels
- [ ] Add the Collaboration Cycle initiative milestone
